java/tools/src/main/java/org/apache/tsfile/tools/TsFileTool.java

```diff
@@ -266,15 +296,57 @@ private static void processDirectory(File directory, ExecutorService executor) {
 
   private static void cpFile(String sourceFilePath, String targetDirectoryPath) {
     try {
-      Files.createDirectories(Paths.get(targetDirectoryPath));
+      String inputFileAbsolutePath = new File(inputDirectoryStr).getAbsolutePath();
+      String soureFlieName = new File(sourceFilePath).getName();
+      String fileOutPutDirStr =
+          targetDirectoryPath
+              + sourceFilePath.replace(inputFileAbsolutePath, "").replace(soureFlieName, "");
+      Files.createDirectories(Paths.get(fileOutPutDirStr));
       Path sourcePath = Paths.get(sourceFilePath);
-      Path targetPath = Paths.get(targetDirectoryPath, sourcePath.getFileName().toString());
+      Path targetPath = Paths.get(fileOutPutDirStr, sourcePath.getFileName().toString());
       Files.copy(sourcePath, targetPath, StandardCopyOption.REPLACE_EXISTING);
     } catch (IOException e) {
       LOGGER.error("Failed to copy file: " + sourceFilePath, e);
     }
   }
 
+  public static void writeToNewCSV(String fileAbsolutePath, List<String> data, String newFileName) {
+
+    if (schema.hasHeader) {
+      try (BufferedReader reader =
+          new BufferedReader(
+              new InputStreamReader(
+                  Files.newInputStream(Paths.get(fileAbsolutePath)), StandardCharsets.UTF_8))) {
+        String line;
+        if ((line = reader.readLine()) != null) {
+          data.add(0, line);
+        }
+      } catch (Exception e) {
+        LOGGER.error("Error reading file: " + e.getMessage());
+      }
+    }
```

Comment:
First, how about caching the header line as a member field in the previous process.
Second, I would assume the size of data to be quite large, so inserting an element to the head may not be efficient if using an ArrayList, and a LinkedList is also inefficient due to the list item overhead. You may just store the header line in a local variable to avoid this.